src/pyodide-py/pyodide/console.py

```diff
@@ -241,6 +248,19 @@ def banner(self):
         build = f"({', '.join(platform.python_build())})"
         return f"Python {version} {build} on WebAssembly VM\n{cprt}"
 
+    def complete(self, source: str) -> List[Any]:
+        """Use jedi to complete a source from local namespace.
+
+        Returns
+        -------
+        A list of Jedi's Completion objects, sorted by name.
+        """
+        try:
+            import jedi
+        except ImportError:
```

Comment:
I think it would be better not to catch the exception and error explicitly. Otherwise when the returned list of exceptions is empty it's a bit hard to debug

Reply:
I agree with this. Just to explain why I designed it like this: if you try to get completion before jedi is available (without awaiting for `preloads_complete`), it will not fail with an error but returns an empty list. I think we can get both actually. I'll try.